javalanglib/src/main/scala/java/lang/Long.scala

```diff
@@ -114,6 +114,46 @@ object Long {
     }
   }
 
+  // Ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/Long.java
+  def toString(value: scala.Long, intRadix: Int): String = {
+    if (intRadix == 10 || intRadix < Character.MIN_RADIX || intRadix > Character.MAX_RADIX) {
+      toString(value)
+    } else if (Integer.MIN_VALUE <= value && (value <= Integer.MAX_VALUE)) {
+      Integer.toString(value.toInt, intRadix)
+    } else {
+      // Character.forDigit checks the digit range which is not the requirement here - hence the apparent duplication
+      def forDigit(digit: Int): Char = {
+        val overBaseTen = digit - 10
+        val result = if (overBaseTen < 0) '0' + digit else 'a' + overBaseTen
+        result.toChar
+      }
+
+      // If the value is zero or positive, we can reduce code by negating the value for the string conversion
+      // but not adding a '-' at the end
+      var _value: scala.Long = if (value >= 0) -value else value
+      val bufSize: Int = 65
+      val buf = Array.ofDim[Char](bufSize)
```

Comment:
I'm pretty sure using a dumb var res: String and prepending is faster on JS (and also more readable). @sjrd?